### PR TITLE
Reimplement scrolling to fragments

### LINKED ISSUE #10753: Intermittent pass in /html/browsers/browsing-the-web/scroll-to-fragid/scroll-to-top.html and /html/browsers/browsing-the-web/scroll-to-fragid/scroll-frag-percent-encoded.html and /html/browsers/browsing-the-web/scroll-to-fragid/scroll-to-id-top.html
```
  ▶ Unexpected subtest result in /html/browsers/browsing-the-web/scroll-to-fragid/scroll-to-top.html:
  └ PASS [expected FAIL] Fragment Navigation: When fragid is TOP scroll to the top of the document
```

```
  ▶ Unexpected subtest result in /html/browsers/browsing-the-web/scroll-to-fragid/scroll-frag-percent-encoded.html:
  └ PASS [expected FAIL] Fragment Navigation: fragment id should be percent-decoded
```

### LINKED ISSUE #13736: Support scrolling to fragment point with webrender.

### PR DESCRIPTION
**Pull request by mrobinson**
---

- [x] `./mach build -d` does not report any errors
- [x] `./mach test-tidy` does not report any errors
- [x] These changes fix #13736, #10753 (github issue number if applicable).

- [x] There are tests for these changes OR
- [ ] These changes do not require tests because _____

This reimplemntation of the feature uses ScrollRootIds to scroll
particular scrollable areas of the page.

Fixes #13736.
Fixes #10753.

---
This change is [Reviewable](https://reviewable.io/reviews/servo/servo/14367)

### CODE REVIEW

**Comment by mrobinson** [comment]
Okay. This is rebased and ready to go.

File: components/script_layout_interface/wrapper_traits.rs [hunk 1]
```diff
@@ -264,6 +264,17 @@ pub trait ThreadSafeLayoutNode: Clone + Copy + Debug + GetLayoutData + NodeInfo
...
     fn get_colspan(&self) -> u32;
+
+    fn scroll_root_id(&self) -> ScrollRootId {
+        let fragment_type = match self.get_pseudo_element_type() {
+            PseudoElementType::Normal => FragmentType::FragmentBody,
+            PseudoElementType::Before(_) => FragmentType::BeforePseudoContent,
+            PseudoElementType::After(_) => FragmentType::AfterPseudoContent,
+            PseudoElementType::DetailsSummary(_) => FragmentType::FragmentBody,
+            PseudoElementType::DetailsContent(_) => FragmentType::FragmentBody,
+        };
```
Comment (pcwalton) [inline components/script_layout_interface/wrapper_traits.rs]: Can you factor this out into a separate function?

**Comment by mrobinson** [comment]
@pcwalton Sure. The latest PR does this.

**Pull request merged**